Python 2.7: 9 syntax errors and 44 undefined names

[flake8](http://flake8.pycqa.org) testing of https://github.com/aiidateam/aiida_core on Python 2.7.14

$ __flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics__
```
./.ci/test_daemon.py:57:22: F821 undefined name 'load_node'
    finished_list = [load_node(pk).is_terminated for pk in pks]
                     ^
./.ci/test_daemon.py:58:18: F821 undefined name 'load_node'
    node_list = [load_node(pk) for pk in pks]
                 ^
./.ci/test_daemon.py:76:62: F821 undefined name 'e'
        print("Note: the command failed, message: {}".format(e))
                                                             ^
./.ci/test_daemon.py:83:16: F821 undefined name 'load_node'
        calc = load_node(pk)
               ^
./.ci/test_daemon.py:116:20: F821 undefined name 'load_node'
            calc = load_node(pk)
                   ^
./.ci/test_daemon.py:166:29: F821 undefined name 'load_node'
            original_calc = load_node(calc.get_extra('_aiida_cached_from'))
                            ^
./.ci/test_daemon.py:239:35: F821 undefined name 'CalculationFactory'
    TemplatereplacerCalculation = CalculationFactory('simpleplugins.templatereplacer')
                                  ^
./.ci/test_daemon.py:290:12: F821 undefined name 'Code'
    code = Code.get_from_string(codename)
           ^
./examples/wf_submission/wf_XTiO3_example.py:29:11: F821 undefined name 'DataFactory'
UpfData = DataFactory('upf')
          ^
./examples/wf_submission/wf_XTiO3_example.py:60:8: F821 undefined name 'NotExistent'
except NotExistent:
       ^
./examples/wf_submission/wf_XTiO3_example.py:67:17: F821 undefined name 'DataFactory'
ParameterData = DataFactory('parameter')
                ^
./examples/work/scratch.py:55:44: F821 undefined name 'StructureData'
        spec.input("structure", valid_type=StructureData)
                                           ^
./examples/work/scratch.py:72:28: F821 undefined name 'Calc'
        return ToContext(r=Calc(1234), s=Outputs(1234))
                           ^
./examples/work/scratch.py:86:44: F821 undefined name 'StructureData'
        spec.input("structure", valid_type=StructureData)
                                           ^
./examples/work/scratch.py:101:33: F821 undefined name 'Calc'
            self.insert_barrier(Calc(pid))
                                ^
./examples/work/scratch.py:105:25: F821 undefined name 'load_node'
            self.out(s, load_node(pid)['output_parameters'].dict)                        ^
./aiida/common/archive.py:90:54: F821 undefined name 'sys'
                          .format(member.name), file=sys.stderr)
                                                     ^
./aiida/common/archive.py:96:54: F821 undefined name 'sys'
                          .format(member.name), file=sys.stderr)
                                                     ^
./aiida/common/orbital/__init__.py:170:12: F821 undefined name 'BaseFactory'
    return BaseFactory('aiida.common.orbital', entry_point)
           ^
./aiida/backends/sqlalchemy/cmdline.py:46:17: F821 undefined name 'timezone'
            t = timezone.now() - datetime.timedelta(days=n_days_ago)
                ^
./aiida/backends/sqlalchemy/cmdline.py:46:34: F821 undefined name 'datetime'
            t = timezone.now() - datetime.timedelta(days=n_days_ago)
                                 ^
./aiida/backends/sqlalchemy/models/workflow.py:240:24: F821 undefined name 'model'
            instance = model(**params)
                       ^
./aiida/backends/sqlalchemy/models/workflow.py:241:13: F821 undefined name 'session'
            session.add(instance)
            ^
./aiida/backends/djsite/db/models.py:975:38: F821 undefined name 'e'
            exc.original_exception = e
                                     ^
./aiida/backends/djsite/db/models.py:1114:38: F821 undefined name 'e'
            exc.original_exception = e
                                     ^
./aiida/backends/tests/work/test_workfunctions.py:145:34: F821 undefined name 'INPUT'
        KWARGS_INPUT = {'b': Int(INPUT)}
                                 ^
./aiida/backends/tests/orm/data/remote.py:52:27: F821 undefined name 'the_path'
                os.remove(the_path)
                          ^
./aiida/utils/queries.py:55:15: F821 undefined name 'InputValidationError'
        raise InputValidationError(
              ^
./aiida/utils/cli/__init__.py:24:9: F821 undefined name 'func'
        func(*args, **kwargs)
        ^
./aiida/utils/cli/__init__.py:24:15: F821 undefined name 'args'
        func(*args, **kwargs)
              ^
./aiida/utils/cli/__init__.py:24:23: F821 undefined name 'kwargs'
        func(*args, **kwargs)
                      ^
./aiida/restapi/translator/data/kpoints.py:137:38: F821 undefined name 'old_label'
                        path.append([old_label, label])
                                     ^
./aiida/orm/importexport.py:2568:39: F821 undefined name 'filename'
                self._zipfile.getinfo(filename)
                                      ^
./aiida/orm/importexport.py:2574:21: F821 undefined name 'filename'
                    filename))
                    ^
./aiida/orm/querybuilder.py:391:45: F821 undefined name 'ormclass'
                    if ormclassifiers[2] != ormclass:
                                            ^
./aiida/orm/querybuilder.py:801:20: F821 undefined name 'edge_exists'
                if edge_exists:
                   ^
./aiida/orm/querybuilder.py:906:72: F821 undefined name 'k'
                                    "".format(', '.join(allowed_keys), k)
                                                                       ^
./aiida/orm/querybuilder.py:1751:37: F821 undefined name 'val'
                        '{}'.format(val, self._tag_to_alias_map.keys())
                                    ^
./aiida/orm/querybuilder.py:1781:23: F821 undefined name 'replacement_dict'
                inp = replacement_dict.get(inp, inp)
                      ^
./aiida/orm/data/array/trajectory.py:203:22: F821 undefined name 'e'
                type(e).__name__, exc))
                     ^
./aiida/scheduler/plugins/direct.py:152:48: F821 undefined name 'virtualMemoryKb'
            lines.append("ulimit -v {}".format(virtualMemoryKb))
                                               ^
./docs/source/concepts/include/snippets/workflows/workchains/run_workchain_keyword.py:6:5: E999 SyntaxError: invalid syntax
    ...
    ^
./docs/source/concepts/include/snippets/workflows/workchains/run_workchain_submit.py:6:5: E999 SyntaxError: invalid syntax
    ...
    ^
./docs/source/concepts/include/snippets/workflows/workchains/run_workchain_get_node_pid.py:6:5: E999 SyntaxError: invalid syntax
    ...
    ^
./docs/source/concepts/include/snippets/workflows/workchains/run_workchain_builder.py:6:5: E999 SyntaxError: invalid syntax
    ...
    ^
./docs/source/concepts/include/snippets/workflows/workchains/run_workchain_expand.py:6:5: E999 SyntaxError: invalid syntax
    ...
    ^
./docs/source/concepts/include/snippets/workflows/workchains/run_workchain_submit_parallel.py:14:31: E999 SyntaxError: invalid syntax
    def submit_workchains(self)
                              ^
./docs/source/concepts/include/snippets/workflows/workchains/run_workchain_submit_append.py:14:31: E999 SyntaxError: invalid syntax
    def submit_workchains(self)
                              ^
./docs/source/concepts/include/snippets/workflows/workchains/run_workchain_submit_internal.py:5:5: E999 SyntaxError: invalid syntax
    ...
    ^
./docs/source/concepts/include/snippets/workflows/workchains/run_workchain_submit_complete.py:14:30: E999 SyntaxError: invalid syntax
    def submit_workchain(self)
                             ^
./docs/source/developer_guide/devel_tutorial/sum_submission.py:15:17: F821 undefined name 'DataFactory'
ParameterData = DataFactory('parameter')
                ^
./docs/source/developer_guide/devel_tutorial/sum_submission.py:35:8: F821 undefined name 'Code'
code = Code.get_from_string(codename)
       ^
./docs/source/developer_guide/devel_tutorial/sum_submission.py:38:12: F821 undefined name 'Computer'
computer = Computer.get(computer_name) 
           ^
9     E999 SyntaxError: invalid syntax
44    F821 undefined name 'load_node'
53
```

As a note: while some are real issues (like 
```
./aiida/backends/djsite/db/models.py:975:38: F821 undefined name 'e'
            exc.original_exception = e
```
(but they should not be removed from the code, rather be fixed - they are inside a exception catching and not tested yet), or some of the errors of the examples, I want to mention that some of the errors for `load_node` and similar (e.g. in `.ci/test_daemon.py` are spurious errors, because that script is run via `verdi run` that pre-loads a number of things, including load_node.

--- a/.ci/test_daemon.py
+++ b/.ci/test_daemon.py
@@ -54,6 +54,7 @@ def print_daemon_log():
 
 
 def jobs_have_finished(pks):
+    global load_node  # instantiated by the verdi run command
     finished_list = [load_node(pk).is_terminated for pk in pks]
     node_list = [load_node(pk) for pk in pks]
     num_finished = len([_ for _ in finished_list if _])
@@ -77,6 +78,7 @@ def print_logshow(pk):
 
 
 def validate_calculations(expected_results):
+    global load_node  # instantiated by the verdi run command
     valid = True
     actual_dict = {}
     for pk, expected_dict in expected_results.items():
@@ -109,6 +111,7 @@ def validate_calculations(expected_results):
 
 
 def validate_workchains(expected_results):
+    global load_node  # instantiated by the verdi run command
     valid = True
     for pk, expected_value in expected_results.items():
         this_valid = True
@@ -144,6 +147,7 @@ def validate_cached(cached_calcs):
     """
     Check that the calculations with created with caching are indeed cached.
     """
+    global load_node  # instantiated by the verdi run command
     valid = True
     for calc in cached_calcs:
 

--- a/aiida/backends/sqlalchemy/cmdline.py
+++ b/aiida/backends/sqlalchemy/cmdline.py
@@ -9,6 +9,8 @@
 ###########################################################################
 
 from __future__ import absolute_import
+import datetime
+from django.utils import timezone
 from aiida.common.log import get_dblogger_extra
 
 

--- a/aiida/common/archive.py
+++ b/aiida/common/archive.py
@@ -10,6 +10,7 @@
 
 from __future__ import absolute_import
 from __future__ import print_function
+import sys
 from six.moves import range
 
 

--- a/docs/source/concepts/include/snippets/workflows/workchains/run_workchain_submit_append.py
+++ b/docs/source/concepts/include/snippets/workflows/workchains/run_workchain_submit_append.py
@@ -11,11 +11,11 @@ class SomeWorkChain(WorkChain):
             cls.inspect_workchains,
         )
 
-    def submit_workchains(self)
+    def submit_workchains(self):
         for i in range(3):
             future = self.submit(SomeWorkChain)
             self.to_context(workchains=append_(future))
 
-    def inspect_workchains(self)
+    def inspect_workchains(self):
         for workchain in self.ctx.workchains:
             assert workchain.is_finished_ok

--- a/docs/source/concepts/include/snippets/workflows/workchains/run_workchain_submit_complete.py
+++ b/docs/source/concepts/include/snippets/workflows/workchains/run_workchain_submit_complete.py
@@ -11,9 +11,9 @@ class SomeWorkChain(WorkChain):
             cls.inspect_workchain,
         )
 
-    def submit_workchain(self)
+    def submit_workchain(self):
         future = self.submit(SomeWorkChain)
         return ToContext(workchain=future)
 
-    def inspect_workchain(self)
+    def inspect_workchain(self):
         assert self.ctx.workchain.is_finished_ok

--- a/docs/source/concepts/include/snippets/workflows/workchains/run_workchain_submit_internal.py
+++ b/docs/source/concepts/include/snippets/workflows/workchains/run_workchain_submit_internal.py
@@ -4,5 +4,5 @@ from aiida.work.workchain import WorkChain
 class AddAndMultiplyWorkChain(WorkChain):
     ...
 
-    def submit_sub_workchain(self)
+    def submit_sub_workchain(self):
         node = self.submit(AddAndMultiplyWorkChain, a=Int(1), b=Int(2), c=Int(3))

--- a/docs/source/concepts/include/snippets/workflows/workchains/run_workchain_submit_parallel.py
+++ b/docs/source/concepts/include/snippets/workflows/workchains/run_workchain_submit_parallel.py
@@ -11,13 +11,13 @@ class SomeWorkChain(WorkChain):
             cls.inspect_workchains,
         )
 
-    def submit_workchains(self)
+    def submit_workchains(self):
         for i in range(3):
             future = self.submit(SomeWorkChain)
             key = 'workchain_{}'.format(i)
             self.to_context(**{key:future})
 
-    def inspect_workchains(self)
+    def inspect_workchains(self):
         for i in range(3):
             key = 'workchain_{}'.format(i)
             assert self.ctx[key].is_finished_ok

--- a/docs/source/developer_guide/devel_tutorial/sum_executable.py
+++ b/docs/source/developer_guide/devel_tutorial/sum_executable.py
@@ -14,7 +14,7 @@ import sys
 in_file = sys.argv[1]
 out_file = sys.argv[2]
 
-print "Some output from the code"
+print("Some output from the code")
 
 with open(in_file) as f:
     in_dict = json.load(f)
@@ -23,4 +23,3 @@ out_dict = { 'sum':in_dict['x1']+in_dict['x2'] }
 
 with open(out_file,'w') as f:
     json.dump(out_dict,f)
-

--- a/docs/source/developer_guide/devel_tutorial/sum_submission.py
+++ b/docs/source/developer_guide/devel_tutorial/sum_submission.py
@@ -8,10 +8,15 @@
 # For further information on the license, see the LICENSE.txt file        #
 # For further information please visit http://www.aiida.net               #
 ###########################################################################
-import sys
-import os
+from __future__ import print_function
 
-from aiida.common.exceptions import NotExistent
+import os
+import sys
+
+from aiida.orm import DataFactory
+from aiida.orm.computer import Computer
+from aiida.orm.implementation.sqlalchemy.code import Code
+
 ParameterData = DataFactory('parameter')
 
 # The name of the code setup in AiiDA
@@ -28,8 +33,8 @@ try:
     else:
         raise IndexError
 except IndexError:
-    print >> sys.stderr, ("The first parameter can only be either "
-                          "--send or --dont-send")
+    print(("The first parameter can only be either "
+                          "--send or --dont-send"), file=sys.stderr)
     sys.exit(1)
 
 code = Code.get_from_string(codename)
@@ -52,12 +57,12 @@ calc.use_parameters(parameters)
 
 if submit_test:
     subfolder, script_filename = calc.submit_test()
-    print "Test submit file in {}".format(os.path.join(
+    print("Test submit file in {}".format(os.path.join(
         os.path.relpath(subfolder.abspath),
         script_filename
-        ))
+        )))
 else:
     calc.store_all()
     calc.submit()
-    print "submitted calculation; calc=Calculation(uuid='{}') # ID={}".format(
-        calc.uuid,calc.dbnode.pk)
+    print("submitted calculation; calc=Calculation(uuid='{}') # ID={}".format(
+        calc.uuid,calc.dbnode.pk))

